Assets/scripts/BattleManager.cs

```diff
@@ -186,6 +192,13 @@ private class PlayerBattleData {
 		battleProgressInMeasures = 0;
 	}
 
+	void Update() {
+		if (isInBattle) {
+			MarkPassedNotes(attacker);

```

Comment:
I'm not sure if it is correct to do this every frame or if I should be doing it just once every beat or something. I made it flexible! Just provide the time since you last called it to the song.getpassednotes method and it'll get all the notes you haven't gotten yet.